2-api/server.js

```diff
@@ -0,0 +1,275 @@
+    return res.status(404).send("Please enter your name to proceed");
+  }
+  pool
+    .query("SELECT name FROM customers WHERE id=$1", [newName])
+    .then((result) => {
+      if (result.rows.length > 0) {
+        return res.status(404).send("Customer name already exists!");
+      } else {
+        pool.query(
+          "INSERT INTO customers (name, address, city, country) VALUES ($1, $2, $3, $4)",
+          [newName, newAddress, newCity, newCountry],
+          (err, result) => {
+            if (!err) {
+              res.status(200).send("Customer Added").json(result.rows);
+            } else {
+              res.status(400);
+              console.error(err);
+            }
+          }
+        );
+      }
+    });
+});
+
+app.delete("/customers/:customerId", (req, res) => {
+  const customerId = req.params.customerId;
+  pool
+    .query("DELETE FROM orders WHERE customer_id=$1", [customerId])
+    .then(() => {
+      pool.query(
+        "DELETE FROM customers WHERE id=$1",
+        [customerId],
+        (err, result) => {
+          if (!err) {
+            res.status(200).send("Success! Customer deleted").json(result.rows);
+          } else {
+            res.status(400).send("Unable to process request");
+            console.error(err);
+          }
+        }
+      );
+    });
+});
+
+app.get("/suppliers", (req, res) => {
+  pool.query("SELECT * FROM suppliers", (err, result) => {
+    if (!err) {
+      res.status(200).json(result.rows);
+    } else {
+      res.status(400);
+      console.error(err);
+    }
+  });
+});
+
+app.delete("/orders/:orderId", (req, res) => {
+  const orderId = req.params.orderId;
+  pool
+    .query("DELETE FROM order_items WHERE order_id=$1", [orderId])
+    .then(() => {
+      pool.query("DELETE FROM orders WHERE id=$1", [orderId], (err, result) => {
+        if (!err) {
+          res.status(200).send("Order deleted successfully").json(result.rows);
+        } else {
+          res.status(400).send("Unable to delete order");
+          console.error(err);
+        }
+      });
+    });
+});
+
+app.get("/products", (req, res) => {
+  const productName = req.query.productName;
+  pool.query(
+    `SELECT product_name, unit_price, supplier_name FROM product_availability INNER JOIN products ON prod_id = products.id INNER JOIN suppliers ON supp_id = suppliers.id ${
+      productName ? "WHERE product_name ILIKE $1" : ""
+    }`,
+    productName ? [`%${productName}%`] : [],
+    (err, result) => {
+      if (!err) {
+        res.status(200).json(result.rows);
+      } else {
+        res.status(400);
+        console.error(err);
+      }
+    }
+  );
+});
+
+app.post("/products", (req, res) => {
+  const newProduct = req.body.newProduct;
+  pool
+    .query("SELECT product_name FROM products WHERE product_name=$1", [
+      newProduct,
+    ])
+    .then((result) => {
+      if (result.rows.length > 0) {
+        return res.status(404).send("Product already exists");
+      } else {
```

Comment:
Good check - you need a different response code - here are some examples https://en.wikipedia.org/wiki/List_of_HTTP_status_codes#4xx_server_errors